src/app/post/[...slug]/page.tsx

```diff
@@ -10,21 +13,54 @@ interface Metadata {
   title?: string;
 }
 
-export default async function Post({ params }: PageProps) {
-  const { slug } = await params;
-  const slugPath = slug.join("/");
+export default function Post({ params }: PageProps) {
+  const paramsData = use(params);
+  const { slug } = paramsData;
+
+  const [metadata, setMetadata] = useState<Metadata | null>(null);
+  const [processedContent, setProcessedContent] = useState<string | null>(null);
+  const [isNotFound, setIsNotFound] = useState<boolean>(false);
+  const [isLoading, setIsLoading] = useState<boolean>(true);
+
+  useEffect(() => {
+    async function fetchData() {
+      setIsLoading(true);
+      setIsNotFound(false);
+      
+      const slugPath = slug.join('/');
+      const contentData = await fetchContent(slugPath);
+      
+      if (contentData === null) {
+        setIsNotFound(true);
+        setIsLoading(false);
+        return;
+      }
+
+      const processedData = await processMarkdown(contentData.content, slugPath);
+      setMetadata(contentData.metadata);
+      setProcessedContent(processedData);
+      setIsLoading(false);
+    }
+
+    fetchData();
+  }, [slug]);
+
+  if (isLoading) {
+    return <div>Loading...</div>;
+  }
 
-  const contentData = await fetchContent(slugPath);
-  if (!contentData) notFound();
+  if (isNotFound) {
+    return <NotFound />;
+  }
 
-  const processedContent = await processMarkdown(contentData.content, slugPath);
-  const metadata: Metadata = contentData.metadata ?? {};
+  if (!metadata || !processedContent) {
+    return <NotFound />;
+  }
```

Comment:
Should we just combine this into one block? Not a significant deal, but it might be cleaner.

```suggestion
  if (isNotFound || !metadata || !processedContent) {
    return <NotFound />;
  }
```

Reply:
combined and updated the code